akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -804,6 +821,9 @@ import scala.util.control.NonFatal
 
     override def onPull(): Unit = externalCallback.invoke(RequestOne)
     override def onDownstreamFinish(cause: Throwable): Unit = externalCallback.invoke(Cancel(cause))
+
+    def createMaterializedTwiceException(): IllegalStateException =
+      new IllegalStateException(s"Substream Source($name) cannot be materialized more than once")
```

Comment:
Perhaps the above function and this in subsource/subsink are better in a common function in Graphstage or more appropriate shared usage error trait, instead of the duplication.

https://github.com/akka/akka/blob/f579b98ecc1e01f158a4ffca990959a82067f6c9/akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala#L745-L746

e.g.
```
def createMaterializedTwiceException(sname: String): IllegalStateException =
  new IllegalStateException(s"Substream $sname cannot be materialized more than once")
```

Reply:
I agree with DRYing up in general but for this there seems to be no good place to put it and the functionality is trivial. I guess a more interesting unification could be between the implementations of `SubSink` or `SubSource` (that have some more duplication) but I'd like to leave that for another time.

Reply:
👍 on not abstracting and creating coupling for something trivial